Add CPU model name to trace files and traceRecord

skrakau opened the pull request:
This adds the CPU model name to the trace files (to allow using this information for a CO2 footprint plugin) and accordingly to the `traceRecord`.

It is not added to any tests yet.

bentsherman commented:
Wondering if we should integrate this with the `arch` directive somehow. See #3786

pditommaso commented on modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy:
```diff
@@ -87,6 +87,7 @@ class TraceRecord implements Serializable {
             syscw:      'num',      // -- /proc/$pid/io
             read_bytes: 'mem',      // -- /proc/$pid/io
             write_bytes:'mem',      // -- /proc/$pid/io
+            cpu_model:  'str',      // -- /proc/cpuinfo field 'model name'
```
Let's add this at the bottom of the list, after `hostname`

pditommaso commented:
I would expect the `arch` to be included in the CPU model

pditommaso reviewed (changes requested):
This looks ok, but there's a failing test. Likey due to new field

skrakau commented:
Hi, 
I moved the new `cpu_model` field to the end of the list, adjusted the `test-bash-wrapper-with-trace.txt` file to the new `command-trace.txt` since this caused an error in the `BashWrapperBuilderTest` test, and added an entry to the docs. I hope I didn't forget anything.

pditommaso commented on docs/tracing.md:
```diff
@@ -307,6 +307,9 @@ The following table shows the fields that can be included in the execution repor
   :::
 : The host on which the task was executed. Supported only for the Kubernetes executor yet. Activate with `k8s.fetchNodeName = true` in the Nextflow config file.
 
+`cpu_model`
+: The name of the CPU model which was used to execute the task. This data is read from file `/proc/cpuinfo`.
```
```suggestion
`cpu_model`
: :::{versionadded} 22.07.0-edge
  :::
: The name of the CPU model used to execute the task. This data is read from file `/proc/cpuinfo`.
```

pditommaso commented on modules/nextflow/src/main/groovy/nextflow/trace/TraceRecord.groovy:
```diff
@@ -100,7 +100,8 @@ class TraceRecord implements Serializable {
             inv_ctxt: 'num',
-            hostname: 'str'
+            hostname: 'str',
+            cpu_model:  'str',      // -- /proc/cpuinfo field 'model name'
```
```suggestion
            cpu_model:  'str'
```

pditommaso commented:
We are ready to go! Thanks for this useful contribution!

skrakau commented:
Thanks for adding it!
